app/database/operator/server_data_operator/handlers/group.ts

```diff
@@ -185,6 +187,75 @@ const GroupHandler = (superclass: any) => class extends superclass implements Gr
 
         return records;
     };
+
+    /**
+     * handleGroupChannelsForChannel: Handler responsible for the Create/Update operations occurring on the GroupChannel table from the 'Server' schema
+     * @param {string} channelId
+     * @param {HandleGroupChannelsForChannelArgs} groupMembershipsArgs
+     * @param {GroupChannel[]} groupMembershipsArgs.groupMemberships
+     * @param {boolean} groupMembershipsArgs.prepareRecordsOnly
+     * @throws DataOperatorException
+     * @returns {Promise<GroupChannelModel[]>}
+     */
+    handleGroupChannelsForChannel = async ({channelId, groups, prepareRecordsOnly = true}: HandleGroupChannelsForChannelArgs): Promise<GroupChannelModel[]> => {
+        // Get existing group memberships
+        const existingGroupChannels = await queryGroupChannelsForChannel(this.database, channelId).fetch();
+
+        let records: GroupChannelModel[] = [];
+        let rawValues: GroupChannel[] = [];
+
+        // Nothing to add or remove
+        if (!groups?.length && !existingGroupChannels.length) {
+            return records;
+        }
+
+        // No groups - remove all existing ones
+        if (!groups?.length && existingGroupChannels.length) {
+            records = existingGroupChannels.map((gm) => gm.prepareDestroyPermanently());
+        }
+
+        // No existing groups - add all new ones
+        if (groups?.length && !existingGroupChannels.length) {
+            rawValues = groups.map((g) => ({id: `${g.id}-${channelId}`, channel_id: channelId, group_id: g.id}));
+        }
+
+        // If both, we only want to save new ones and delete one's no longer in groups
+        if (groups?.length && existingGroupChannels.length) {
+            const groupsSet: {[key: string]: GroupChannel} = {};
+
+            for (const g of groups) {
+                groupsSet[`${g.id}`] = {id: `${g.id}-${channelId}`, channel_id: channelId, group_id: g.id};
```

Comment:
What is the difference between ` ${g.id}` and just using `g.id`?

Reply:
Ah typescript used to complain a few versions ago about indexing by string on Array's, old habit - have updated.